daemon/daemon_linux.go

```diff
@@ -69,7 +70,21 @@ func (daemon *Daemon) cleanupMountsFromReaderByID(reader io.Reader, id string, u
 
 // cleanupMounts umounts shm/mqueue mounts for old containers
 func (daemon *Daemon) cleanupMounts() error {
-	return daemon.cleanupMountsByID("")
+	if err := daemon.cleanupMountsByID(""); err != nil {
+		return err
+	}
+
+	infos, err := mount.GetMounts()
+	if err != nil {
+		return errors.Wrap(err, "error reading mount table for cleanup")
+	}
+
+	info := getMountInfo(infos, daemon.root)
+	if info == nil || info.Root != daemon.root {
+		// daemon root is not a mountpoint, nothing to clean up
+		return nil
+	}
+	return mount.Unmount(daemon.root)
```

Comment:
Bit confused here; the function description mentions "umounts shm/mqueue mounts for old containers", but here we're unmounting `/var/lib/docker`?

Reply:
Updated the doc string.